tests/generate_test_data.py

```diff
@@ -73,6 +85,21 @@ async def main(user, password):
                 fh.write(json.dumps(device_buckets, indent=2))
 
 
+def _recursive_data_anonymize(data: str | dict | list) -> dict:
+    if isinstance(data, list):
+        for elem in data:
+            _recursive_data_anonymize(elem)
+
+    elif isinstance(data, dict):
+        for elem in data.keys():
+            if elem in ANONYMIZE_ATTRIBUTES:
+                data[elem] = "ANONYMIZED"
```

Comment:
I think hashing the values might be a better approach, because the same deviceId used in different parts of the API responses would lead to the same hashed value, without leaking any personal data.

Maybe sth like this would enough? https://stackoverflow.com/a/4567109